src/Raven.Client/Documents/AI/Chat.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Documents.Operations.AI.Agents;
+using Raven.Client.Util;
+
+namespace Raven.Client.Documents.AI;
+internal class Chat<T> : IChatOperations<T> where T : new()
+{
+    private readonly AiOperations _aiOperations;
+    private readonly string _agent;
+    private readonly Dictionary<string, object> _scope;
+
+    private string _chatId;
+    private AiUsage _totalUsage;
+    private List<ToolRequest> _toolsRequests;
+    private List<ToolResponse> _toolsResponses = [];
+    private string _userPrompt;
+
+    private bool _firstRun = true;
+    public Chat(AiOperations aiOperations, string agent, Dictionary<string, object> scope)
+    {
+        _aiOperations = aiOperations;
+        _agent = agent;
+        _scope = scope;
+    }
+    public Chat(AiOperations aiOperations, string chatId)
+    {
+        _aiOperations = aiOperations;
+        _chatId = chatId;
+    }
+
+    public IEnumerable<ToolRequest> OpenTools() => _toolsRequests ?? throw new InvalidOperationException("You have to call RunAsync first");
+
+    public void AddToolResponse(string id, string content)
+    {
+        _toolsResponses.Add(new ToolResponse
+        {
+            ToolId = id,
+            Content = content
+        });
+    }
+
+    public bool Run() => AsyncHelpers.RunSync(() => RunAsync(CancellationToken.None));
+
+    public void SetPrompt(string userPrompt)
+    {
+        _userPrompt = userPrompt;
+    }
+
+    private T _answer;
+    public T Answer => _answer ?? throw new InvalidOperationException("You have to call RunAsync first");
+    public string Id => _chatId ?? throw new InvalidOperationException("This is a new chat, the ID wasn't set yet, you have to call RunAsync");
+    public AiUsage TotalUsage => _totalUsage ?? throw new InvalidOperationException("You have to call RunAsync first");
+
+    public async Task<IEnumerable<ChatMessage>> ReadMessagesAsync(CancellationToken token)
+    {
+        var id = _chatId ?? throw new InvalidOperationException("This is a new chat, the ID wasn't set yet, you have to call RunAsync.");
+        using var session = _aiOperations._store.OpenAsyncSession();
+        var d = await session.LoadAsync<ChatDocument>(id, token).ConfigureAwait(false);
+        return d.Messages;
```

Comment:
remember that user might have different deserialization conventions, we should not use session and user conventions DocumentConventions.Default should be used